docs/core-concepts/route-params.md

```diff
@@ -136,27 +134,88 @@ const idFormatParam: ParamGetSet<IdFormat> = {
 If you define your path params with a question mark, `:?` the router assumes this param is not required.
 
 ```ts
-const routes = [
+const routes = createRoutes([
   {
     name: 'users',
     path: '/users/:?id', // [!code focus]
     component: ...
   }
-] as const satisfies Routes
+])
 ```
 
 This means the route will match even if nothing is provided after `/users/`.
 
 This also works for non-string params.
 
 ```ts
-const routes = [
+const routes = createRoutes([
   {
     name: 'users',
     path: path('/users/:?id', { id: Number }), // [!code focus]
     component: ...
   }
-] as const satisfies Routes
+])
 ```
 
 Which now, `router.params.id` has the type `number | undefined`, and will only match url where the value passes as a number or is missing entirely.
+
+## Param Name
+
+There are very few constraints on the name you choose for a param. Your param name can include any special character except forward slash `/`. Kitbag Router uses forward slash (or the end of the string) to denote the end of a param name in a path.
+
+```ts
+const routes = createRoutes([
+  {
+    name: 'users',
+    path: '/users/:can-have#what.ever!?',
```

Comment:
Im curious if that's even valid for a url. My guess is you can't use a # or a ? Because those mean the start of a hash or query.

Reply:
that's just the name of the param, not the value that will end up in a url

Reply:
oh right 👍